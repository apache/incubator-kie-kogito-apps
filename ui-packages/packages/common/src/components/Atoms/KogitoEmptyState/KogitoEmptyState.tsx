import React from 'react';
import {
  Bullseye,
  Button,
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStateVariant,
  Title
} from '@patternfly/react-core';
import {
  ExclamationTriangleIcon,
  InfoCircleIcon,
  SearchIcon
} from '@patternfly/react-icons';
import '@patternfly/patternfly/patternfly-addons.css';
import { OUIAProps, componentOuiaProps } from '../../../utils/OuiaUtils';

export enum KogitoEmptyStateType {
  Search,
  Refresh,
  Reset,
  Info
}

interface IOwnProps {
  type: KogitoEmptyStateType;
  title: string;
  body: string;
  onClick?: () => void;
}

export const KogitoEmptyState: React.FC<IOwnProps & OUIAProps> = ({
  type,
  title,
  body,
  onClick,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <Bullseye {...componentOuiaProps(ouiaId, 'kogito-empty-state', ouiaSafe)}>
      <EmptyState variant={EmptyStateVariant.full}>
        {type === KogitoEmptyStateType.Search && (
          <EmptyStateIcon icon={SearchIcon} size="sm" />
        )}
        {(type === KogitoEmptyStateType.Refresh ||
          type === KogitoEmptyStateType.Reset) && (
          <EmptyStateIcon
            icon={ExclamationTriangleIcon}
            size="sm"
            color="var(--pf-global--warning-color--100)"
          />
        )}
        {type === KogitoEmptyStateType.Info && (
          <EmptyStateIcon
            icon={InfoCircleIcon}
            size="sm"
            color="var(--pf-global--info-color--100)"
          />
        )}

        <Title headingLevel="h5" size="lg">
          {title}
        </Title>

        <EmptyStateBody>{body}</EmptyStateBody>

        {type === KogitoEmptyStateType.Refresh && (
          <Button variant="primary" onClick={onClick}>
            Refresh
          </Button>
        )}

        {type === KogitoEmptyStateType.Reset && (
          <Button variant="link" onClick={onClick}>
            Reset to default
          </Button>
        )}
      </EmptyState>
    </Bullseye>
  );
};
