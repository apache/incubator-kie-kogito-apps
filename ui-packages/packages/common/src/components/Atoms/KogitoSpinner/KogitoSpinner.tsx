import React from 'react';
import {
  Title,
  EmptyState,
  EmptyStateIcon,
  Spinner
} from '@patternfly/react-core';
import { OUIAProps, componentOuiaProps } from '../../../utils/OuiaUtils';

interface KogitoSpinnerProps {
  spinnerText: string;
}
const KogitoSpinner: React.FC<KogitoSpinnerProps & OUIAProps> = ({
  spinnerText,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <EmptyState {...componentOuiaProps(ouiaId, 'kogito-spinner', ouiaSafe)}>
      <EmptyStateIcon variant="container" component={Spinner} />
      <Title size="lg">{spinnerText}</Title>
    </EmptyState>
  );
};

KogitoSpinner.displayName = 'KogitoSpinner';
export default KogitoSpinner;
