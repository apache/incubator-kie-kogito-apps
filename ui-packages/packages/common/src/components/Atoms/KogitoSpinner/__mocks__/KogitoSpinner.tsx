import React from 'react';

const MockedKogitoSpinner = () => {
  return <></>;
};

export default MockedKogitoSpinner;
