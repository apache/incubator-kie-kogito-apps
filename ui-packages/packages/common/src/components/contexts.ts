import React from 'react';

export const aboutLogoContext = React.createContext('');
