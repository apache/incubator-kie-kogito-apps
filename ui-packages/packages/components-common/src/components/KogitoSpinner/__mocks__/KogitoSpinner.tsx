import React from 'react';

const MockedKogitoSpinner: React.FC = () => {
  return <></>;
};

export default MockedKogitoSpinner;
