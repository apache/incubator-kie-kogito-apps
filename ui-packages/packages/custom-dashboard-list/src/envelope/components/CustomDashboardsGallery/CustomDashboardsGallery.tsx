/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { OUIAProps, componentOuiaProps } from '@kogito-apps/ouia-tools';
import { Bullseye, Gallery, GalleryItem } from '@patternfly/react-core';
import { CustomDashboardInfo } from '../../../api/CustomDashboardListEnvelopeApi';
import CustomDashboardCard from '../CustomDashboardCard/CustomDashboardCard';
import { CustomDashboardListDriver } from '../../../api/CustomDashboardListDriver';
import {
  KogitoEmptyState,
  KogitoEmptyStateType,
  KogitoSpinner
} from '@kogito-apps/components-common';

export interface CustomDashboardGalleryProps {
  driver: CustomDashboardListDriver;
  customDashboardsDatas: CustomDashboardInfo[];
  isLoading: boolean;
}

const CustomDashboardsGallery: React.FC<
  CustomDashboardGalleryProps & OUIAProps
> = ({ driver, customDashboardsDatas, isLoading, ouiaId, ouiaSafe }) => {
  if (isLoading) {
    return (
      <Bullseye>
        <KogitoSpinner
          spinnerText="Loading customDashboard..."
          ouiaId="custom-dashboard-list-loading-custom-dashboard"
        />
      </Bullseye>
    );
  }

  if (
    !isLoading &&
    customDashboardsDatas &&
    customDashboardsDatas.length === 0
  ) {
    return (
      <KogitoEmptyState
        type={KogitoEmptyStateType.Search}
        title="No results found"
        body="Try using different filters"
      />
    );
  }

  return (
    <Gallery
      hasGutter
      style={{ margin: '25px' }}
      {...componentOuiaProps(ouiaId, 'customDashboard-gallery', ouiaSafe)}
    >
      {customDashboardsDatas &&
        customDashboardsDatas.map((customDashboardData, index) => (
          <GalleryItem key={index}>
            <CustomDashboardCard
              customDashboardData={customDashboardData}
              key={index}
              driver={driver}
            ></CustomDashboardCard>
          </GalleryItem>
        ))}
    </Gallery>
  );
};

export default CustomDashboardsGallery;
