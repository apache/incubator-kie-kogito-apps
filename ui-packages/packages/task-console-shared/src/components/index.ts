export { default as TaskState } from './TaskState/TaskState';
