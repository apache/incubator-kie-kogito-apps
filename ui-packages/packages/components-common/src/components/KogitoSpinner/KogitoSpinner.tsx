import React from 'react';
import {
  EmptyState,
  EmptyStateIcon
} from '@patternfly/react-core/dist/js/components/EmptyState';
import { Title } from '@patternfly/react-core/dist/js/components/Title';
import { Spinner } from '@patternfly/react-core/dist/js/components/Spinner';
import {
  OUIAProps,
  componentOuiaProps
} from '@kogito-apps/ouia-tools/dist/utils/OuiaUtils';

interface KogitoSpinnerProps {
  spinnerText: string;
}
export const KogitoSpinner: React.FC<KogitoSpinnerProps & OUIAProps> = ({
  spinnerText,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <EmptyState {...componentOuiaProps(ouiaId, 'kogito-spinner', ouiaSafe)}>
      <EmptyStateIcon variant="container" component={Spinner} />
      <Title size="lg" headingLevel="h3">
        {spinnerText}
      </Title>
    </EmptyState>
  );
};
