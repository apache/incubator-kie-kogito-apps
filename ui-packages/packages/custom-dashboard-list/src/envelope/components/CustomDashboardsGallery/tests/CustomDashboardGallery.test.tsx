/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';
import CustomDashboardsGallery from '../CustomDashboardsGallery';
import { render, screen } from '@testing-library/react';
import {
  customDashboardInfos,
  MockedCustomDashboardListDriver
} from '../../../tests/mocks/MockedCustomDashboardsListDriver';
import { KogitoSpinner } from '@kogito-apps/components-common/dist/components/KogitoSpinner';
import { KogitoEmptyState } from '@kogito-apps/components-common/dist/components/KogitoEmptyState';

const MockedComponent = (): React.ReactElement => {
  return <></>;
};

describe('customDashboard gallery tests', () => {
  Date.now = jest.fn(() => 1487076708000);
  const driver = new MockedCustomDashboardListDriver();
  it('renders gallery of customDashboard', () => {
    const { container } = render(
      <CustomDashboardsGallery
        driver={driver}
        isLoading={false}
        customDashboardsDatas={customDashboardInfos}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders loading component', () => {
    const { container } = render(
      <CustomDashboardsGallery
        driver={driver}
        isLoading={true}
        customDashboardsDatas={customDashboardInfos}
      />
    );
    expect(container).toMatchSnapshot();
    const checkLoading = screen.getByText('Loading customDashboard...');
    expect(checkLoading).toBeTruthy();
  });

  it('renders empty state component', () => {
    const { container } = render(
      <CustomDashboardsGallery
        driver={driver}
        isLoading={false}
        customDashboardsDatas={[]}
      />
    );
    expect(container).toMatchSnapshot();
    const checkEmptyState = screen.getByText('No results found');
    expect(checkEmptyState).toBeTruthy();
  });
});
