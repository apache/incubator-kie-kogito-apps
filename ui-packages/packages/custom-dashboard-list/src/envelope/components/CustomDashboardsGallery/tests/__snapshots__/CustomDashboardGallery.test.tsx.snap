// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customDashboard gallery tests renders empty state component 1`] = `
<div>
  <div
    class="pf-l-bullseye"
    data-ouia-component-type="kogito-empty-state"
    data-ouia-safe="true"
  >
    <div
      class="pf-c-empty-state"
    >
      <div
        class="pf-c-empty-state__content"
      >
        <svg
          aria-hidden="true"
          class="pf-c-empty-state__icon"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
        <h5
          class="pf-c-title pf-m-lg"
          data-ouia-component-id="OUIA-Generated-Title-2"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          No results found
        </h5>
        <div
          class="pf-c-empty-state__body"
        >
          Try using different filters
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`customDashboard gallery tests renders gallery of customDashboard 1`] = `
<div>
  <div
    class="pf-l-gallery pf-m-gutter"
    data-ouia-component-type="customDashboard-gallery"
    data-ouia-safe="true"
    style="margin: 25px;"
  >
    <div>
      <article
        class="pf-c-card pf-m-selectable"
        data-ouia-component-id="OUIA-Generated-Card-1"
        data-ouia-component-type="PF4/Card"
        data-ouia-safe="true"
        data-testid="card"
        id=""
        tabindex="0"
      >
        <div
          class="pf-c-card__header"
        >
          <div
            class=""
          >
            Empty
          </div>
        </div>
        <div
          class="pf-c-card__header"
        >
          <h1
            class="pf-u-font-weight-bold"
            data-ouia-component-id="OUIA-Generated-Text-1"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            dashboard1
          </h1>
        </div>
        <div
          class="pf-c-card__body"
        >
          <div
            class="pf-u-mt-md"
          >
            <form
              class="pf-c-form"
              novalidate=""
            >
              <div
                class="pf-c-form__group"
              >
                <div
                  class="pf-c-form__group-label"
                >
                  <label
                    class="pf-c-form__label"
                    for="path"
                  >
                    <span
                      class="pf-c-form__label-text"
                    >
                      Path
                    </span>
                  </label>
                   
                </div>
                <div
                  class="pf-c-form__group-control"
                >
                  <p
                    class=""
                    data-ouia-component-id="OUIA-Generated-Text-2"
                    data-ouia-component-type="PF4/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    /user/home
                  </p>
                  
                </div>
              </div>
              <div
                class="pf-c-form__group"
              >
                <div
                  class="pf-c-form__group-label"
                >
                  <label
                    class="pf-c-form__label"
                    for="lastModified"
                  >
                    <span
                      class="pf-c-form__label-text"
                    >
                      LastModified
                    </span>
                  </label>
                   
                </div>
                <div
                  class="pf-c-form__group-control"
                >
                  <p
                    class=""
                    data-ouia-component-id="OUIA-Generated-Text-3"
                    data-ouia-component-type="PF4/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    <time
                      datetime="1657564200000"
                    >
                      in 5 years
                    </time>
                  </p>
                  
                </div>
              </div>
            </form>
          </div>
        </div>
      </article>
    </div>
    <div>
      <article
        class="pf-c-card pf-m-selectable"
        data-ouia-component-id="OUIA-Generated-Card-2"
        data-ouia-component-type="PF4/Card"
        data-ouia-safe="true"
        data-testid="card"
        id=""
        tabindex="0"
      >
        <div
          class="pf-c-card__header"
        >
          <div
            class=""
          >
            Empty
          </div>
        </div>
        <div
          class="pf-c-card__header"
        >
          <h1
            class="pf-u-font-weight-bold"
            data-ouia-component-id="OUIA-Generated-Text-4"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            dashboard2
          </h1>
        </div>
        <div
          class="pf-c-card__body"
        >
          <div
            class="pf-u-mt-md"
          >
            <form
              class="pf-c-form"
              novalidate=""
            >
              <div
                class="pf-c-form__group"
              >
                <div
                  class="pf-c-form__group-label"
                >
                  <label
                    class="pf-c-form__label"
                    for="path"
                  >
                    <span
                      class="pf-c-form__label-text"
                    >
                      Path
                    </span>
                  </label>
                   
                </div>
                <div
                  class="pf-c-form__group-control"
                >
                  <p
                    class=""
                    data-ouia-component-id="OUIA-Generated-Text-5"
                    data-ouia-component-type="PF4/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    /user/home
                  </p>
                  
                </div>
              </div>
              <div
                class="pf-c-form__group"
              >
                <div
                  class="pf-c-form__group-label"
                >
                  <label
                    class="pf-c-form__label"
                    for="lastModified"
                  >
                    <span
                      class="pf-c-form__label-text"
                    >
                      LastModified
                    </span>
                  </label>
                   
                </div>
                <div
                  class="pf-c-form__group-control"
                >
                  <p
                    class=""
                    data-ouia-component-id="OUIA-Generated-Text-6"
                    data-ouia-component-type="PF4/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    <time
                      datetime="1657564200000"
                    >
                      in 5 years
                    </time>
                  </p>
                  
                </div>
              </div>
            </form>
          </div>
        </div>
      </article>
    </div>
  </div>
</div>
`;

exports[`customDashboard gallery tests renders loading component 1`] = `
<div>
  <div
    class="pf-l-bullseye"
  >
    <div
      class="pf-c-empty-state"
      data-ouia-component-id="custom-dashboard-list-loading-custom-dashboard"
      data-ouia-component-type="kogito-spinner"
      data-ouia-safe="true"
    >
      <div
        class="pf-c-empty-state__content"
      >
        <div
          class="pf-c-empty-state__icon"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-xl"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <h3
          class="pf-c-title pf-m-lg"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Loading customDashboard...
        </h3>
      </div>
    </div>
  </div>
</div>
`;
